src/notebooks/pipes/Data/BucketSelector.tsx

```diff
@@ -64,10 +61,10 @@ const BucketSelector: FC = () => {
     )
   }
 
-  if (loading === RemoteDataState.Done && !selectedBucketName) {
+  if (loading === RemoteDataState.Done && !data.bucket) {
     buttonText = 'Choose a bucket'
-  } else if (loading === RemoteDataState.Done && selectedBucketName) {
-    buttonText = selectedBucketName
+  } else if (loading === RemoteDataState.Done && data.bucket) {
+    buttonText = data.bucket.name
```

Comment:
```suggestion
    buttonText = data.bucket?.name
```

Reply:
handled by line 66 in the conditional right above it